### ISSUE TITLE
vars dictionary yields unevaluated literal expression

### ISSUE THREAD

**Issue by dmrzzz** [ansible/ansible#45585]
##### SUMMARY

The [FAQ item regarding dynamic variable names](https://docs.ansible.com/ansible/latest/reference_appendices/faq.html#when-should-i-use-also-how-to-interpolate-variables-or-dynamic-variable-names) seems to imply that `{{ vars['FOO'] }}` is the same as `{{ FOO }}`, but that doesn't appear to be the case when FOO is set to an expression.

##### ISSUE TYPE
- Bug Report

##### ANSIBLE VERSION

```paste below
ansible 2.6.4
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/dmrz/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/dmrz/.local/lib/python3.5/site-packages/ansible
  executable location = /home/dmrz/.local/bin/ansible
  python version = 3.5.2 (default, Nov 23 2017, 16:37:01) [GCC 5.4.0 20160609]
```

##### CONFIGURATION

(none)

##### OS / ENVIRONMENT

Linux host (xubuntu 16.04.4)

##### STEPS TO REPRODUCE

```yaml
- hosts: localhost
  gather_facts: no
  vars:

    FOO: "{{ lookup('env', 'FOO') }}"

  tasks:

    # interpolates as expected
    - name: normal variable reference
      debug: msg={{ FOO }}

    # yields literal lookup expression?!
    - name: reference using vars dictionary
      debug: msg={{ vars['FOO'] }}

    # always skips (not what I want)
    - name: verify required variables, should fail if FOO unset
      fail: msg="Required variable '{{ item }}' not provided"
      when: not vars[item]
      loop:
        - FOO
```

```bash
$ unset FOO
$ ansible-playbook varsdict.yml
$ export FOO=bar
$ ansible-playbook varsdict.yml
```

##### EXPECTED RESULTS

With FOO unset, both debug tasks should print an empty msg and the fail task should fail.
With FOO=bar, both debug tasks should print msg "bar" and the fail task should be skipped.

##### ACTUAL RESULTS

The first debug task behaves as expected, but the second debug task using the vars dictionary expands to the unevaluated literal expression, and the fail task is _always_ skipped because the unevaluated literal expression (not its value) is truthy.

```paste below
dmrz@golbez:~/tmp/ansible-test$ unset FOO
dmrz@golbez:~/tmp/ansible-test$ ansible-playbook varsdict.yml 
 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'


PLAY [localhost] **************************************************************************************************************************************

TASK [normal variable reference] **********************************************************************************************************************
ok: [localhost] => {
    "msg": ""
}

TASK [reference using vars dictionary] ****************************************************************************************************************
ok: [localhost] => {
    "msg": "{{ lookup('env', 'FOO') }}"
}

TASK [verify required variables, should fail if FOO unset] ********************************************************************************************
skipping: [localhost] => (item=FOO) 

PLAY RECAP ********************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0   

dmrz@golbez:~/tmp/ansible-test$ export FOO=bar
dmrz@golbez:~/tmp/ansible-test$ ansible-playbook varsdict.yml 
 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'


PLAY [localhost] **************************************************************************************************************************************

TASK [normal variable reference] **********************************************************************************************************************
ok: [localhost] => {
    "msg": "bar"
}

TASK [reference using vars dictionary] ****************************************************************************************************************
ok: [localhost] => {
    "msg": "{{ lookup('env', 'FOO') }}"
}

TASK [verify required variables, should fail if FOO unset] ********************************************************************************************
skipping: [localhost] => (item=FOO) 

PLAY RECAP ********************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0   
```

**Comment by bcoca** [comment]
fixed via #42412

**Comment by dmrzzz** [comment]
Cool, thanks!

For anyone else who finds this issue, the solution is to replace `{{ vars['FOO'] }}` with `{{ lookup('vars', 'FOO') }}`
